2/src/components/__tests__/ListItem.spec.js

```diff
@@ -0,0 +1,88 @@
+﻿import { shallowMount } from '@vue/test-utils'
+import ListItem from '../ListItem.vue'
+
+const factory = (values = {}) => {
+  return shallowMount(ListItem, {
+    propsData : values
+  })
+}
+
+describe('ListItem', () => {
+	describe('Given none empty todo', () => {
+		const wrapper = factory({
+				todo: { id: '1', message: 'Hello', },
+		})
+
+		it('Renders an edit button', () => {
+			const editButton = wrapper.find('.editButton')
+
+			expect(editButton.exists()).toBeTruthy()
+			expect(editButton.text()).toContain('Edit');
+		})
+
+		it('Renders a delete button', () => {
+			const deleteButton = wrapper.find('.deleteButton')
+
+			expect(deleteButton.exists()).toBeTruthy()
+			expect(deleteButton.text()).toContain('Delete');
+		})
+
+		it('Emits delete-todo event after click on delete button', () => {
+			const deleteButton = wrapper.find('.deleteButton')
+			deleteButton.trigger('click')
+
+			expect(wrapper.emitted('delete-todo')).toBeTruthy();
+		})
+
+		it('Renders the todo text', () => {
+			expect(wrapper.text()).toContain('Hello');
+		})
+
+		describe('After clicking edit button', () => {
+			beforeAll(() => {
```

Comment:
if you would re-initialize your wrapper you could use `beforeEach` instead of `beforeAll`.